eis_toolkit/prediction/cnn_classification_and_probability.py

```diff
@@ -0,0 +1,417 @@
+from typing import Literal, Union
+
+import numpy as np
+import pandas as pd
+import sklearn.preprocessing
+import tensorflow as tf
+from beartype import beartype
+from keras import Model
+from pandas import DataFrame
+from sklearn.metrics import confusion_matrix
+from sklearn.preprocessing import OneHotEncoder, StandardScaler
+from sklearn.utils.class_weight import compute_sample_weight
+
+from eis_toolkit.exceptions import CNNException, CNNRunningParameterException, InvalidArgumentTypeException
+from eis_toolkit.prediction.model_performance_estimation import performance_model_estimation
+
+
+def normalize_the_data(scaler_agent: sklearn.preprocessing, data: np.ndarray):
+    """
+    Normalize multi-dimensional data using a specified scaler.
+    
+    This function applies a scaling technique to each feature in the data, transforming the data to a specified range or distribution. It is particularly useful for normalizing image data or similarly structured multi-dimensional data.
+
+    Parameters:
+        - scaler_agent: An instance of a preprocessing scaler from sklearn.preprocessing (e.g., MinMaxScaler, StandardScaler) that will be used to apply normalization.
+        - data: A NumPy ndarray containing the data to be normalized. The data is expected to be in a multi-dimensional format, typically including dimensions for sample size, height, width, and channels (e.g., for image data).
+
+    Returns:
+        - normalized_data: The input data normalized according to the scaler_agent. The output data maintains the original shape of the input data.
+        
+    """
+    number_of_sample, h, w, c = data.shape
+    temp = scaler_agent.transform(data.reshape(-1, data.shape[-1]))
+    normalized_data = temp.reshape(number_of_sample, h, w, c)
+    return normalized_data
```

Comment:
If you want to implement a normalizing function, please put it into its own file and under `transformations` module. Also, we don't want users to have to deal with any sklearn classes if possible, so don't put any such stuff into parameters. Let's keep it as functional style and modular as possible.

Please also add return type annotation, change the one letter variable names to something more descriptive and convert the docstring to match EIS Toolkit conventions/style. I would also consider renaming this function to "normalize".

I should point out still that it's not mandatory to implement this function as part of this PR.

Reply:
ok perfect I put it just in case, if it is not good we just delete it :-)